File: ListAlloc/list_alloc.js
function allocate() {
    data = new Array(1000).fill(0);
    return data;
}

results = [];
for (var i = 0; i < 125; ++i) {
    var t0 = process.hrtime();
    allocate();
    var stop = process.hrtime(t0);
    results.push(stop[1] / 1000000);  // process.hrtime(start_time) returns [seconds, nanoseconds]
}

mean = 0;
for (var i = 0; i < 125; ++i) {
    mean += results[i];
}
mean /= 125;

console.log(`Mean time ${mean} ms`);

results.sort();

if (results.length % 2 == 0)
    console.log(`Median time: ${(results[Math.floor((results.length / 2) - 1)] + results[Math.floor(results.length / 2)]) / 2.0} ms`);
else
    console.log(`Median time: ${results[Math.floor(results.length / 2)]} ms`);

temp = 0;
for (var i = 0; i < 125; ++i) {
    temp += (results[i] - mean) * (results[i] - mean);
}
stddev = Math.sqrt(temp / 125);

console.log(`Stddev: ${stddev} ms`);

File: ListAlloc/list_alloc.rb
def allocate
  data = Array.new(1000){0}
  return data
end


results = []
125.times do |i|
  start = Time.new
  allocate
  stop = Time.new - start
  results << stop * 1000  # Time.new is in seconds
end
mean = results.inject(:+) / results.size
puts("Mean time: #{mean}ms")

results = results.sort
median = 0
if results.size % 2 == 0
  median = results[results.size.to_i / 2 - 1] + results[results.size.to_i / 2]
  median /= 2
else
  median = results[results.size.to_i / 2]
end
puts("Median time: #{median}ms")

stddev = (results.inject(0){|save, a| save + (a - mean) ** 2} / 125) ** 0.5
puts("Stddev: #{stddev}ms")